frontend/src/lib/utils/d3Utils.ts

```diff
@@ -0,0 +1,69 @@
+import * as d3 from 'd3'
+import { INITIAL_CONFIG } from 'scenes/insights/Histogram/histogramUtils'
+import { D3Selector, D3Transition } from 'lib/hooks/useD3'
+
+export const getOrCreateEl = (
+    container: D3Selector,
+    selector: string,
+    createCallback: () => D3Selector
+): D3Selector => {
+    const el = container.select(selector)
+    if (el.empty()) {
+        return createCallback()
+    }
+    return el
+}
+
+export const animate = (
+    it: D3Selector,
+    transitionDuration: number,
+    isAnimated: boolean,
+    toAnimate: (_it: D3Transition | D3Selector) => D3Transition | D3Selector = (_it) => _it // everything you want to animate goes here
+): D3Transition | D3Selector => {
+    if (isAnimated) {
+        return it.transition().duration(transitionDuration).call(toAnimate)
+    }
+    return it.call(toAnimate)
+}
+
+export const wrap = (
+    text: D3Selector,
+    width: number,
+    lineHeight: number = INITIAL_CONFIG.spacing.labelLineHeight
+): D3Selector => {
+    const maxWidth = width - 6 // same as padding-{left|right}: 3px;
+    text.each(function () {
+        const _text = d3.select(this)
+        console.log(_text)
```

Comment:
did you mean to leave this console log? :o

Reply:
![image](https://user-images.githubusercontent.com/13460330/126723469-e03b504c-e9c5-4ac1-9498-404cbbae7501.png)